cmd/registry/plugins/registry-lint-openapi-sample/main.go

```diff
@@ -0,0 +1,215 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/apex/log"
+	lint "github.com/apigee/registry/cmd/registry/plugins/linter"
+	"github.com/apigee/registry/rpc"
+	"gopkg.in/yaml.v3"
+)
+
+// sampleOpenApiLintCommandExecuter is an interface through which the Sample OpenAPI linter executes.
+type sampleOpenApiLintCommandExecuter interface {
+	// Runs the sample OpenAPI linter with a provided spec and configuration path
+	Execute(specPath string, ruleIds []string) ([]*rpc.LintProblem, error)
+}
+
+type DescriptionField struct {
+	LineNumber   int
+	ColumnNumber int
+	Description  string
+}
+
+// sampleOpenApiLinterRunner implements the LinterRunner interface for the sample OpenAPI linter.
+type sampleOpenApiLinterRunner struct{}
+
+// concreteSampleOpenApiLintCommandExecuter implements the sampleOpenApiLintCommandExecuter interface
+// for the sample OpenAPI linter.
+type concreteSampleOpenApiLintCommandExecuter struct{}
+
+const descriptionLessThan1000CharsRuleId = "description-less-than-1000-chars"
+const descriptionContainsNoTagsRuleId = "description-contains-no-tags"
+
+func (linter *sampleOpenApiLinterRunner) Run(req *rpc.LinterRequest) (*rpc.LinterResponse, error) {
+	return linter.RunImpl(req, &concreteSampleOpenApiLintCommandExecuter{})
+}
+
+func (linter *sampleOpenApiLinterRunner) RunImpl(
+	req *rpc.LinterRequest,
+	executer sampleOpenApiLintCommandExecuter,
+) (*rpc.LinterResponse, error) {
+	lintFiles := make([]*rpc.LintFile, 0)
+
+	// Traverse the files in the directory
+	err := filepath.Walk(req.GetSpecDirectory(), func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Async API and Open API specs are YAML files
+		if !strings.HasSuffix(path, ".yaml") && !strings.HasSuffix(path, ".yml") {
+			return nil
+		}
+
+		// Execute the linter.
+		problems, err := executer.Execute(path, req.GetRuleIds())
+		if err != nil {
+			return err
+		}
+
+		// Formulate the response.
+		lintFiles = append(lintFiles, &rpc.LintFile{
+			FilePath: path,
+			Problems: problems,
+		})
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &rpc.LinterResponse{
+		Lint: &rpc.Lint{
+			Name:  "registry-lint-openapi-sample",
+			Files: lintFiles,
+		},
+	}, nil
+}
+
+func (*concreteSampleOpenApiLintCommandExecuter) Execute(specPath string, ruleIds []string) ([]*rpc.LintProblem, error) {
```

Comment:
From above, I think this should just be a function (not an instance method). I also think the name "Execute" isn't very informative. "lintFile" seems to describe what this does and would be one better name for this IMO.